CRM/Member/BAO/MembershipType.php

```diff
@@ -128,6 +128,20 @@ public static function add(&$params, $ids = array()) {
     if ($id) {
       self::updateAllPriceFieldValue($id, $params);
     }
+
+    if (!empty($params['custom']) &&
+      is_array($params['custom'])
+    ) {
+      CRM_Core_BAO_CustomValueTable::store($params['custom'], 'civicrm_membership_type', $membershipType->id);
+    }
+
+    if (!empty($id)) {
+      CRM_Utils_Hook::post('edit', 'MembershipType', $membershipType->id, $membershipType);
+    }
+    else {
+      CRM_Utils_Hook::post('create', 'MembershipType', $membershipType->id, $membershipType);
+    }
```

Comment:
@colemanw I've added hook_civicrm_post support into MembershipType here as I saw it was not being called - does that seem correct to you?

Reply:
@mattwire I feel like rather than add support to the add functions (which would then be called twice via the api) we should call the api from the form

Reply:
Yes, makes sense

Reply:
@eileenmcnaughton But the API (basic create) still doesn't call the post (or pre) hooks so not sure that quite works.  I can see the value of calling the API instead of MembershipType::add though, just think we still need the pre/post hooks in the add function, and that means we also need the customvalue code in the add function (the API would do this for us but would be after the post hook).